File: 0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        int n=nums.length;
        if(n==0){
            return new ArrayList<>();
        }
        
        int i=1;
        List<String>ans=new ArrayList<>();
        while(i<=n){
            int count=1;
            int prev=nums[i-1];
            while(i<n && nums[i]==prev+1){
                prev=nums[i];
                i++;
                count++;
            }
            if(count==1){
                ans.add(String.valueOf(prev));
                
            }else{
                ans.add(nums[i-count]+"->"+nums[i-1]);
            }
            i++;
        }
        return ans;
    }
}

File: 0228-summary-ranges/0228-summary-ranges.py

class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        n = len(nums)
        if n == 0:
            return []

        i = 1
        ans = []
        while i <= n:
            count = 1
            prev = nums[i - 1]
            while i < n and nums[i] == prev + 1:
                prev = nums[i]
                i += 1
                count += 1
            if count == 1:
                ans.append(str(prev))
            else:
                ans.append(f"{nums[i - count]}->{nums[i - 1]}")
            i += 1
        return ans
